XSCALE-Alliance/XSCALE-Alliance.github.io
# Update gems

Status:

 * most gems updated to a recent version; most notably all Middleman gems
 * gem `sprockets` for now fixed on v3.7.2, upgrading to v4 breaks
 * upgraded to Ruby 2.5

To verify:

 * make sure you have Ruby 2.5 available on your system (e.g. `rvm install 2.5`)
 * open the repo in a terminal 
 * verify your Ruby version `ruby --version`; should be 2.5
 * run `make bootstrap` to install and/or update the gems locally

Verify that:

 * [x] `make build-verbose` or `make-build` exits without error code
 * [x] `make server` shows a beautiful website, without errors
 * [x] the github action [`verify_pull_request`](https://github.com/XSCALE-Alliance/XSCALE-Alliance.github.io/actions?query=workflow%3Averify_pull_request) should pass
 * [ ] the `make deploy` target should work - @zefram-cochrane please verify this
 * [x] verify that all github actions have ruby 2.5 specified (I might have missed this)



<details>

Observations:

A blunt `bundle update` results in the following error:

```
Bundler could not find compatible versions for gem "bundler":
  In Gemfile:
    middleman (~> 4.1.1) was resolved to 4.1.1, which depends on
      middleman-core (= 4.1.1) was resolved to 4.1.1, which depends on
        bundler (~> 1.1)
```

So, I update to a recent version of middleman by simply dropping the version constraint on it.

stuff seems to break when upgrading from middleman-sprockets 4.0.0.rc.1

</details>
